aten/src/ATen/core/dispatch/OperatorEntry.cpp

```diff
@@ -220,7 +220,15 @@ void OperatorEntry::updateDispatchTableEntry_(const c10::Dispatcher& dispatcher,
 }
 
 void OperatorEntry::updateDispatchTable_(const c10::Dispatcher& dispatcher, DispatchKey dispatch_key) {
-  for (auto k : c10::getRuntimeDispatchKeys(dispatch_key)) {
+  // Undefined isn't a runtime key but we have an entry for it in dispatchTable_ (which is not necessary).
+  // Update it separately if required.
+  // XXX: in fact I think we should just start from iter=1 in updateDispatchTableFull_.
```

Comment:
cc: @ezyang Here's the special handling for `Undefined` I mentioned offline. ;)

Reply:
OK.  Long term, I actually think having the entry for Undefined in the dispatch table is useful, as (1) it gives people place to specify functionality that should run when there are no dispatch keys, e.g., an empty TensorList argument and (2) it would let us remove the explicit error checking code in the dispatch hotpath, and so when no dispatch keys are available we just slide into the undefined handler which would then raise the error message.

But indeed, this is awkward if you want to represent RuntimeDispatchKeySet as a DispatchKeySet, as is done here. I think what we should probably do is just say that alias keys cannot validly include undefined tensor. (Except maybe the math key? Hmm...)